lib/src/controller/database.dart

```diff
@@ -1,13 +1,500 @@
-import 'package:path/path.dart';
+import 'dart:convert';
+import 'dart:io';
+import 'package:flutter/material.dart' show ThemeMode;
+import 'package:drift_flutter/drift_flutter.dart';
+import 'package:flex_color_scheme/flex_color_scheme.dart';
+import 'package:interstellar/src/api/comments.dart';
+import 'package:interstellar/src/api/feed_source.dart';
+import 'package:interstellar/src/controller/feed.dart';
+import 'package:interstellar/src/controller/server.dart';
+import 'package:interstellar/src/controller/profile.dart';
+import 'package:interstellar/src/controller/filter_list.dart';
+import 'package:interstellar/src/models/post.dart';
+import 'package:interstellar/src/screens/feed/feed_screen.dart';
+import 'package:interstellar/src/widgets/actions.dart';
+import 'package:interstellar/src/utils/utils.dart';
+import 'package:oauth2/oauth2.dart';
 import 'package:path_provider/path_provider.dart';
+import 'package:drift/drift.dart';
+import 'package:sembast/sembast.dart';
 import 'package:sembast/sembast_io.dart';
+import 'package:path/path.dart';
+
+part 'database.g.dart';
+
+class CredentialsConverter extends TypeConverter<Credentials, String>
+    with JsonTypeConverter2<Credentials, String, String> {
+  const CredentialsConverter();
+
+  @override
+  Credentials fromSql(String fromDb) {
+    return Credentials.fromJson(jsonDecode(fromDb));
+  }
+
+  @override
+  String toSql(Credentials credentials) {
+    return jsonEncode(credentials);
+  }
+
+  @override
+  Credentials fromJson(String json) {
+    return Credentials.fromJson(json);
+  }
+
+  @override
+  String toJson(Credentials value) {
+    return jsonEncode(value);
+  }
+}
+
+class Accounts extends Table {
```

Comment:
I'm not sure how well cascades work with Drift, but if we add a primary key here and then use foreign key references in other locations in the database, we could automatically update things when the account is deleted (like the auto switch account setting).